### PR TITLE
combine network info & add optimism

### PR DESCRIPTION
**Pull request by namgold**
Signed-off-by: Nam Nguyen <nam.nguyen@kyber.network>

### CODE REVIEW

File: src/pages/About/AboutKyberSwap.tsx [hunk 1]
```diff
@@ -1031,12 +1035,14 @@ function AboutKyberSwap() {
               />
               <VelasLogoFull color={isDarkMode ? undefined : 'black'} />
               <AuroraFull />
+              <div />
```
Comment (viet-nv) [inline src/pages/About/AboutKyberSwap.tsx:1038]: why do we need an empty div here?

File: src/pages/ProAmmPool/index.tsx [hunk 2]
```diff
@@ -123,7 +123,7 @@ export default function ProAmmPool() {
               </Flex>
             </Flex>
 
-            <ExternalLink href={`${PROMM_ANALYTICS}/${CHAIN_ROUTE[chainId as ChainId]}/account/${account}`}>
+            <ExternalLink href={`${PROMM_ANALYTICS_URL[chainId as ChainId]}/account/${account}`}>
```
Comment (viet-nv) [inline src/pages/ProAmmPool/index.tsx:126]: PROMM_ANALYTICS_URL now includes  "chain route", right?

Comment (viet-nv) [reply src/pages/ProAmmPool/index.tsx:126]: /ethereum
/bnb
....

File: src/state/lists/updater.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export default function Updater(): null {
 
   // get all loaded lists, and the active urls
   const lists = useAllLists()
-  const activeListUrls = useActiveListUrls()
+  const activeListUrls = useActiveListUrls() //todo: check if this is needed?
```
Comment (viet-nv) [inline src/state/lists/updater.ts:21]: remember to resolve this todo

File: src/state/pair/types.ts [hunk 4]
```diff
@@ -1,10 +0,0 @@
-export interface PopularPair {
```
Comment (viet-nv) [inline src/state/pair/types.ts:1]: this isn't used anymore, is this?

File: src/utils/avaxMainnetTokenMapping.ts [hunk 5]
```diff
@@ -1,15 +1,14 @@
 import { ZERO_ADDRESS } from 'constants/index'
 import { isAddress } from 'utils'
-import { AVAX_MAINNET_TOKEN_LIST } from 'constants/tokenLists/avax.mainnet.tokenlist'
 
 export const getAvaxMainnetTokenLogoURL = (address: string) => {
-  let uri = AVAX_MAINNET_TOKEN_LIST.tokens.find(item => item.address.toLowerCase() === address.toLowerCase())?.logoURI
+  let uri
```
Comment (viet-nv) [inline src/utils/avaxMainnetTokenMapping.ts:5]: this will be affect token image, please check again careful

File: src/utils/dmm.ts [hunk 6]
```diff
@@ -502,24 +503,7 @@ export function useRewardTokensFullInfo(): Token[] {
...
-      : chainId && [25, 338].includes(chainId)
-      ? 'CRO'
-      : chainId && chainId === ChainId.BTTC
-      ? 'BTT'
-      : chainId && chainId === ChainId.VELAS
-      ? 'VLX'
-      : chainId && chainId === ChainId.OASIS
-      ? 'ROSE'
-      : 'ETH'
+  const nativeName = NETWORKS_INFO[chainId || ChainId.MAINNET].nativeToken.symbol
```
Comment (viet-nv) [inline src/utils/dmm.ts:506]: I'm thinking about nativeOnChain function 
maybe that func and this config is duplicate, can we use nativeOnChain in network config?

File: src/utils/ethereumMainnetTokenMapping.ts [hunk 7]
```diff
@@ -27,10 +26,6 @@ export const getEthereumMainnetTokenLogoURL = (address: string) => {
     uri = 'https://i.imgur.com/zhZo9Bx.jpg'
   }
 
-  if (!uri) {
```
Comment (viet-nv) [inline src/utils/ethereumMainnetTokenMapping.ts:30]: same here, some token image will be affect

File: src/utils/index.ts [hunk 8]
```diff
@@ -570,76 +476,50 @@ export const getTokenLogoURL = (inputAddress: string, chainId?: ChainId): string
 
   let imageURL
 
+  imageURL = store
```
Comment (viet-nv) [inline src/utils/index.ts:479]: wow, default image was moved to this. nice job bro

**Review by viet-nv — CHANGES_REQUESTED** [review]

File: src/constants/lists.ts [hunk 9]
```diff
@@ -25,7 +26,13 @@ const ARBITRUM = 'https://bridge.arbitrum.io/token-list-42161.json'
 export const UNSUPPORTED_LIST_URLS: string[] = [BA_LIST]
 export const BYPASS_LIST = [SPOOKY, ARBITRUM]
 // lower index == higher priority for token import
+export const KyberTokensList = (Object.keys(NETWORKS_INFO).map(Number) as ChainId[]).map(
+  chainId => NETWORKS_INFO[chainId].tokenListUrl,
+)
+
+// order matter
```
[the remark this answers is not in the document]
Comment (namgold) [reply src/constants/lists.ts:33]: image
it reflects directly to UI here

[the remark this answers is not in the document]
Comment (viet-nv) [reply src/constants/lists.ts:33]: cool. Can you add Kyber Network logo for it?

File: src/components/TradingViewChart/datafeed.tsx [hunk 10]
```diff
@@ -41,6 +41,8 @@ const getNetworkString = (chainId: ChainId | undefined) => {
       return 'chain-aurora'
     case ChainId.OASIS:
       return 'chain-oasis'
+    case ChainId.OPTIMISM:
+      return 'chain-optimism' //todo namgold: optimism fill this
```
[the remark this answers is not in the document]
Comment (namgold) [reply src/components/TradingViewChart/datafeed.tsx:45]: Yeah, check with @XiaoYhun and it is corrected. Will remove comment now

[the remark this answers is not in the document]
Comment (viet-nv) [reply src/components/TradingViewChart/datafeed.tsx:45]: then please remove todo comment here

File: src/constants/networks/avax.ts [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+      factory: '0x1c758aF0688502e49140230F6b0EBd376d429be5',
+    },
+    dynamic: {
+      zap: '0x83D4908c1B4F9Ca423BEE264163BC1d50F251c31',
+      router: '0x8Efa5A9AD6D594Cf76830267077B78cE0Bc5A5F8',
+      factory: '0x10908C875D865C66f271F5d3949848971c9595C9',
+    },
+    routerV2: '0x00555513Acf282B42882420E5e5bA87b44D8fA6E',
+    aggregationExecutor: '0x41684b361557E9282E0373CA51260D9331e518C9',
+    migrate: EMPTY,
```
[the remark this answers is not in the document]
Comment (namgold) [reply src/constants/networks/avax.ts]: removed all

File: src/state/pair/types.ts (hunk 4 above)
Comment (namgold) [reply src/state/pair/types.ts:1]: Yea, no reference found and no error when removing :D

File: src/utils/index.ts (hunk 8 above)
Comment (namgold) [reply src/utils/index.ts:479]: Yes, we find in tokenlist first, if not, switch case will be the fallback logic. We can actually remove switch case logic in the future.

File: src/utils/avaxMainnetTokenMapping.ts (hunk 5 above)
Comment (namgold) [reply src/utils/avaxMainnetTokenMapping.ts:5]: This was moved to outside here
https://github.com/KyberNetwork/dmm-interface/pull/1034/files#diff-6147e3c1761df71fd40952dbcbac388a45f80b8c318f367ef41048351a2c0c99R479-R485

File: src/utils/ethereumMainnetTokenMapping.ts (hunk 7 above)
Comment (namgold) [reply src/utils/ethereumMainnetTokenMapping.ts:30]: same above, I moved it to outside
[/ethereumMainnetTokenMapping.ts](https://github.com/KyberNetwork/dmm-interface/pull/1034/files/c62b48348836654f8f1e95ef2b786d30d60d047a#diff-e95cc20307d5762e34a7594551a6ca8ede388ae09617edee29fb095c85150a94)

File: src/utils/dmm.ts (hunk 6 above)
Comment (namgold) [reply src/utils/dmm.ts:506]: Yea it duplicate in someway. We may do a refactor for it later.

File: src/pages/About/AboutKyberSwap.tsx (hunk 1 above)
Comment (namgold) [reply src/pages/About/AboutKyberSwap.tsx:1049]: to make it 5-5-(div)3 layout

File: src/hooks/useApproveCallback.ts [hunk 12]
```diff
@@ -164,5 +163,5 @@ export function useProAmmApproveCallback(
     () => (trade && trade.inputAmount.currency.isToken ? trade.maximumAmountIn(allowedSlippage) : undefined),
     [trade, allowedSlippage],
   )
-  return useApproveCallback(amountToApprove, !!chainId ? PRO_AMM_ROUTERS[chainId] : undefined)
+  return useApproveCallback(amountToApprove, !!chainId ? NETWORKS_INFO[chainId].elastic.routers : undefined)
```
[the remark this answers is not in the document]
Comment (namgold) [reply src/hooks/useApproveCallback.ts:166]: I named it folowing `PRO_AMM_ROUTERS`,

**Pull request merged**